engine/consensus/dkg/reactor_engine.go

```diff
@@ -161,34 +229,67 @@ func (e *ReactorEngine) EpochSetupPhaseStarted(currentEpochCounter uint64, first
 	e.registerPhaseTransition(curDKGInfo.phase3FinalView, dkgmodule.Phase3, e.end(nextEpochCounter))
 }
 
-// EpochCommittedPhaseStarted handles the EpochCommittedPhaseStarted protocol event. It
-// compares the key vector locally produced by Consensus nodes against the FlowDKG smart contract
-// key vectors. If the keys don't match a log statement will be invoked. In the happy case the locally
-// produced key should match, if the keys do not match the node will have a invalid random beacon key.
-func (e *ReactorEngine) EpochCommittedPhaseStarted(currentEpochCounter uint64, first *flow.Header) {
+// handleEpochCommittedPhaseStarted checks that the local DKG completed and
+// that our locally computed key share is consistent with the canonical key vector.
+//
+// If our key is inconsistent, we log a warning message.
+// CAUTION: we will still sign using the inconsistent key, to be fixed with Crypto
+// v2 by falling back to staking signatures in this case.
+//
+func (e *ReactorEngine) handleEpochCommittedPhaseStarted(currentEpochCounter uint64) {
+
+	// the epoch counter the DKG just completed is for
+	dkgEpochCounter := currentEpochCounter + 1
+
+	log := e.log.With().
+		Uint64("cur_epoch", currentEpochCounter). // the epoch we are in the middle of
+		Uint64("dkg_for_epoch", dkgEpochCounter). // the epoch the just-finished DKG was preparing for
+		Logger()
+
+	// TODO - first check whether we've already stored the dkg end state, since
+	// we need to handle multiple calls to this method
+
+	// TODO execute in transaction scope of protocol event emitter, need to use
+	// lower-level storage access here (since State doesn't allow tx injection)
 	nextDKG, err := e.State.Final().Epochs().Next().DKG()
 	if err != nil {
-		e.log.Err(err).Msg("checking DKG key consistency: could not retrieve next DKG info")
+		// CAUTION: this should never happen, indicates a storage failure or corruption
+		log.Fatal().Err(err).Msg("checking beacon key consistency: could not retrieve next DKG info")
 		return
 	}
 
-	dkgPrivInfo, err := e.keyStorage.RetrieveMyBeaconPrivateKey(currentEpochCounter + 1)
-	if err != nil {
-		e.log.Err(err).Msg("checking DKG key consistency: could not retrieve DKG private info for next epoch")
+	myBeaconPrivKey, err := e.dkgState.RetrieveMyBeaconPrivateKey(currentEpochCounter + 1)
```

Comment:
Is it possible this method would return `(nil, nil)`? 

If DKG was completed, but a node has no key generated, would it store `nil` as beacon private key, if so, [myBeaconPrivKey.PublicKey()](https://github.com/onflow/flow-go/pull/1721/files#diff-97d28bb0b296d39ca2f3c1a1c2cf3c34db65aa0e992efbadc14f0148dd764fe7R279) would panic.

Reply:
Good point - @tarakby is it possible for this line to return `nil`?

> I'll update `DKGState` to explicitly disallow that case anyway.

Reply:
Yes that's possible, I noticed that case but I'm waiting for @jordanschalm's third DKG PR to look at everything together. 

It is the point I addressed in this PR https://github.com/onflow/flow-go/pull/1718 and Jordan mentioned he will pick it up into his upcoming PR.

Reply:
https://github.com/onflow/flow-go/pull/1721/commits/a87622afa1e7f96c0a4cedbdd79e13e4392a5480